managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedCursorTest.java

```diff
@@ -229,6 +229,28 @@ void readTwice() throws Exception {
         entries.forEach(Entry::release);
     }
 
+    @Test
+    void testPersistentMarkDeleteIfCreateCursorLedgerFailed() throws Exception {
```

Comment:
The test covered the case that the cursor didn't have a Ledger before. Could you please help also cover the case that the cursor had a Ledger before? I think it will not work because the old Ledger will not be removed if the new Ledger creation is failed. When the broker loads the cursor again, it will try to read from the old Ledger first, which will skip all the updates that happened on the metadata store.

Reply:
> I think it will not work because the old Ledger will not be removed if the new Ledger creation is failed. 

Thank you for reminding me, this is an important point. Since the state of the cursor will change to `NoLedger` after the new Ledger creation is failed.  Cursor will try to persist the information to ZK directly. So it will work. 

> The test covered the case that the cursor didn't have a Ledger before. Could you please help also cover the case that the cursor had a Ledger before?

Sure, I added the new test.

Reply:
Ah, got it. The cursor Ledger ID is updated to -1 after persisting to ZooKeeper.